data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+[Thread-0] INFO io.confluent.devrel.datacontracts.shared.BaseConsumer - Received Membership 22781e70-8d2c-4d02-b325-9cdc8663ed19, {"user_id": "22781e70-8d2c-4d02-b325-9cdc8663ed19", "start_date": "2022-06-10", "end_date": "2025-11-11"}
+[Thread-0] INFO io.confluent.devrel.datacontracts.shared.BaseConsumer - Received Membership 06ec5d95-d38c-4bac-9693-a01596eeedd7, {"user_id": "06ec5d95-d38c-4bac-9693-a01596eeedd7", "start_date": "2024-03-25", "end_date": "2025-02-25"}
+[Thread-0] INFO io.confluent.devrel.datacontracts.shared.BaseConsumer - Received Membership 881bf838-1fe1-423e-9afc-0742e3080b5b, {"user_id": "881bf838-1fe1-423e-9afc-0742e3080b5b", "start_date": "2022-07-25", "end_date": "2025-07-08"}
+[Thread-0] INFO io.confluent.devrel.datacontracts.shared.BaseConsumer - Received Membership 73254321-72f9-4783-949f-652f5ca9542e, {"user_id": "73254321-72f9-4783-949f-652f5ca9542e", "start_date": "2022-05-11", "end_date": "2025-03-11"}
+```
+
+</details>
+
+#### Evolving to Version 2
+
+The decision is made to refactor the `membership` schema to encapsulate the date fields into a type - `ValidityPeriod` - which can be reused in other event types.
+
+```avroschema
+{
+    "type": "record",
+    "name": "ValidityPeriod",
+    "fields": [
+      {"name": "from", "type": {"type": "int", "logicalType": "date"}},
+      {"name": "to", "type": {"type": "int", "logicalType": "date"}
+      }
+    ]
+}
+```
+
+The refactored `membership` schema now references this new type:
+
+```avroschema
+{
+  "name": "Membership",
+  "namespace": "io.confluent.devrel",
+  "type": "record",
+  "fields": [
+    { "name": "user_id", "type": "string" },
+    { "name": "validity_period", "type": "io.confluent.devrel.ValidityPeriod" }
+  ]
+}
+```
+
+When we register version 2 of the `membership-avro-value` subject, use `addLocalReference` to include the `validityPeriod` type:
+
+```kotlin
+schemaRegistry{
+    register {
+        subject(inputSubject =  "membership-avro-value", type = "AVRO", file = "$avroSchemaDir/membership_v1.avsc")
+            .setMetadata("$metadataDir/membership_major_version_1.json")
+
+        subject(inputSubject =  "membership-avro-value", type = "AVRO", file = "$avroSchemaDir/membership_v2.avsc")
+            .addLocalReference("validity-period", "$avroSchemaDir/validity_period.avsc")
+            .setMetadata("$metadataDir/membership_major_version_2.json")
+            .setRuleSet("$rulesetDir/membership_migration_rules.json")
+    }
+}
+```
+
+Register the new schema using our gradle plugin:
+
+```shell
+./gradlew :data-contracts:schemas:registerSchemasTask
+```
+
+Download the schema to the version 2 application and generate the Java classes for this schema:
+
+```shell
+./gradlew :data-contracts:app-schema-v2:generateCode
+```
+
+In the `app-schema-v2` module, we'll find a new implementation of a `MembershipConsumer`:
+
+```kotlin
```

Comment:
similar feedback to above - suggest removing or trimming the code snippets and console output down. it's a little overwhelming and I think distracts from the flow of running through it.